### PR TITLE
feat: Webhook for work

### PR DESCRIPTION
**Pull request by Arvindthiru**
### Description of your changes

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### Special notes for your reviewer

#513 needs to be merged to use BeforeEach and AfterEach sections introduced

### CODE REVIEW

File: pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	workv1alpha1 "sigs.k8s.io/work-api/pkg/apis/v1alpha1"
```
Comment (ryanzhang-oss) [inline pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go:21]: FYI, we have already moved the work to our own.

File: pkg/webhook/validation/uservalidation.go [hunk 2]
```diff
@@ -82,27 +84,25 @@ func ValidateMemberClusterUpdate(currentObj, oldObj client.Object, whiteListedUs
 
 // ValidateInternalMemberClusterUpdate checks to see if user is allowed to update argued internal member cluster resource.
 func ValidateInternalMemberClusterUpdate(ctx context.Context, client client.Client, imc fleetv1alpha1.InternalMemberCluster, whiteListedUsers []string, userInfo authenticationv1.UserInfo, subResource string) admission.Response {
-	imcKind := "InternalMemberCluster"
 	namespacedName := types.NamespacedName{Name: imc.Name, Namespace: imc.Namespace}
 	if subResource == "status" {
```
Comment (ryanzhang-oss) [inline pkg/webhook/validation/uservalidation.go:88]: as I mentioned in a previous PR, this logic should be in the handle function, it's just an overhead.

File: pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook_test.go [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+
+const (
+	resourceAllowedFormat                = "user: %s in groups: %v is allowed to modify resource %s: %+v"
+	resourceDeniedFormat                 = "user: %s in groups: %v is not allowed to modify resource %s: %+v"
+	resourceStatusUpdateNotAllowedFormat = "user: %s in groups: %v is not allowed to update %s status: %+v"
+	resourceAllowedGetMCFailed           = "user: %s in groups: %v is allowed to updated %s: %+v because we failed to get MC"
+)
+
+func TestHandleInternalMemberCluster(t *testing.T) {
+	mockClient := &test.MockClient{
```
Comment (ryanzhang-oss) [inline pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook_test.go:31]: why not make this a field so we can test failure cases?

Comment (Arvindthiru) [reply pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook_test.go:31]: The mock client is being passed as a field within the request object and I declared it here to reduce LOC where I saw the same client is being reused and I am testing a failure case as well

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**